Jenkinsfile_CNP

```diff
@@ -1,13 +1,164 @@
 #!groovy
-
+import uk.gov.hmcts.contino.AppPipelineDsl
+properties(
+        [
+                [
+                        $class       : 'GithubProjectProperty',
+                        projectUrlStr: 'https://github.com/hmcts/civil-sdt-commissioning'
+                ],
+                pipelineTriggers([[$class: 'GitHubPushTrigger']])
+        ]
+)
 @Library("Infrastructure")
 
+import uk.gov.hmcts.contino.GradleBuilder
+
 def type = "java"
 def product = "civil"
 def component = "sdt-commissioning"
 
+GradleBuilder builder = new GradleBuilder(this, product)
+
+def secrets = [
+        'civil-${env}' : [
+//                secret('OAUTH2-CLIENT-SECRET', 'OAUTH2_CLIENT_SECRET'),
+//                secret('OAUTH2-CLIENT-ID', 'OAUTH2_CLIENT_ID'),
+//                secret('LD-SDK-KEY', 'LD_SDK_KEY')
+        ]
+]
+
+static LinkedHashMap<String, Object> secret(String secretName, String envVar) {
+  [$class     : 'AzureKeyVaultSecret',
+   secretType : 'Secret',
+   name       : secretName,
+   version    : '',
+   envVariable: envVar
+  ]
+}
+
+def vaultOverrides = [
+        'preview' : 'aat',
+        'spreview': 'saat'
+]
+
+// Configure branches to sync with master branch
+def branchesToSync = ['demo', 'ithc', 'perftest']
+
 withPipeline(type, product, component) {
+  overrideVaultEnvironments(vaultOverrides)
+//  loadVaultSecrets(secrets)
+  enableDbMigration('civil')
   enableAksStagingDeployment()
-  disableLegacyDeployment()
-  nonServiceApp()
+  syncBranchesWithMaster(branchesToSync)
+
+  env.IDAM_URL = "https://idam-api.aat.platform.hmcts.net"
+
+  onPR() {
+    echo "Loading PR secrets"
+//    loadVaultSecrets(prSecrets)
+    env.ENV='preview'
+  }
+
+  onNonPR() {
+    echo "Loading secrets"
+//    loadVaultSecrets(secrets)
+    env.ENV='aat'
+  }
+
+  after('akschartsinstall') {
+
+    echo "Just waiting a while to ensure that the pod has run the job"
+    sh "sleep 120s"
+
+    // Execute functional tests
+    echo "Verifying that functional test run correctly"
+    try {
+      builder.gradle('functional')
+    } finally {
+      junit '**/test-results/**/*.xml'
+    }
+  }
+
+  after('sonarscan') {
```

Comment:
The "after" method is deprecated - see master build pipeline for message about it.  This will need to be changed to one of 'alwaysAfter', 'afterSuccess' or 'afterFailure' (most likely 'alwaysAfter').  All occurrences of it in this file will need to be changed.